src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -722,12 +722,22 @@ bool ControlBoardWrapper::attachAll(const PolyDriverList &polylist)
 
     for(int p=0;p<polylist.size();p++)
     {
-        // find appropriate entry in list of subdevices
-        // and attach
+        // look if we have to attach to a calibrator
+        std::string tmpKey=polylist[p]->key.c_str();
+        if(tmpKey == "Calibrator" || tmpKey == "calibrator")
+        {
+            // Set the IRemoteCalibrator interface, the wrapper must point to the calibrato rdevice
+            yarp::dev::IRemoteCalibrator *calibrator;
+            polylist[p]->poly->view(calibrator);
+
+            IRemoteCalibrator::setCalibratorDevice(calibrator);
```

Comment:
this branch is specific to "Calibrators"; if you change
IRemoteCalibrator::setCalibratorDevice(calibrator);

to IRemoteCalibrator::attach(PolyDriver &);

we could remove this special case?